pkg/cmd/cli/cmd/env.go

```diff
@@ -260,7 +265,10 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 				return nil
 			}
 			for _, c := range containers {
-				c.Env = updateEnv(c.Env, env, remove)
+				c.Env, envUpdated = updateEnv(c.Env, env, remove)
+				if envUpdated {
```

Comment:
all the items in `updated` array are set to `false` by default, so no point to check for it, just put the `envUpdated` value to it.